### PR TITLE
doc: update authentication and grant consent diagrams

### PR DESCRIPTION
**Pull request by kleyow**
:thinking: Added the invalid OTP flow to diagram.

Unsure if I need to add more edge case errors like

- dfsp backend returns invalid payload which doesn't conform to 
```
{ isValid: true|false }
``` 
- No scopes being returned from dfsp backend.

Since we don't validate backend requests payloads atm.

### CODE REVIEW

**Comment by lewisdaly** [comment]
thanks Kevin. Can you please make this a _separate_ puml file, and explicitly add a new section to the linking document (maybe section 4, _Linking Error Scenarios_)? That makes this documentation much more readable.

File: docs/linking/4-grant-consent.puml [hunk 1]
```diff
@@ -47,5 +51,36 @@ Switch -> PISP ++: ""POST /consents""\n\
     ""}""
 
 PISP --> Switch: ""202 Accepted""
+deactivate Switch
+deactivate PISP
+
+== Failure case ==
+
+DFSP -> DFSP: Unable to fetch stored scopes
```
Comment (lewisdaly) [inline docs/linking/4-grant-consent.puml:59]: I'm not sure I understand this scenario. Maybe if you split it into it's own sequence diagram, that responds with a `PUT /consentRequests/456/error` to a `PATCH /consentRequests/456` after some generic processing error on the DFSP side, that would make more sense.

**Comment by kleyow** [comment]
@lewisdaly lmk if this is better.

**Comment by lewisdaly** [comment]
Yep, thanks @kleyow

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**